psycopg/psycopg/rows.py

```diff
@@ -211,6 +211,32 @@ def kwargs_row__(values: Sequence[Any]) -> T:
     return kwargs_row_
 
 
+def scalar_row(index: int = 0) -> BaseRowFactory[Any]:
```

Comment:
Is there really an use case for the index to be something else than 0?

In my understanding this row factory can be used every time you have a query selecting a single row and you want to get a list with all the result, or you run a query and you want a single number as a result.

Is there really a practical case in which someone may run a query wanting the second column as result, discarding all the rest?

I propose to make this a simple row factory, to use as

```python
conn.cursor(row_factory=scalar_row)
```

and always return the first column.

If you think that the index is something that should be specified:

- the index may be optional, in the sense that the factory could be called both as `row_factory=scalar_row` and `row_factory=scalar_row(0)`. This can be implemented the same way decorators that may or may not take arguments are implemented: at the outermost call you can check whether the function is called with an index or with a cursor and act accordingly.
- If we accept an index, it should also be possible to specify a column name, such as `row_factory=scalar_row("id")`.

Reply:
I don't see a practical case running a query containing more columns than one but returning only a scalar value. Such queries can be easily rewritten to return only one column. In this case, I prefer simplicity over non-practical functionality.